dask/dataframe/core.py

```diff
@@ -312,6 +312,7 @@ def __init__(self, dsk, name, meta, divisions):
             )
         self._meta = meta
         self.divisions = tuple(divisions)
+        self._lens = lens
```

Comment:
Should there be any validation? Like does it need to be the same len as divisions - 1?